_tests/managed_tests/README.md

```diff
@@ -1,42 +1,176 @@
-    - Dispatch a QIT test with `--wait` and `--json` parameters
-    - Compare the JSON result with a previously saved snapshot
+4. **Snapshot Verification**
+  - Each test’s final JSON output is compared against a **snapshot** with PHPUnit.
+  - Mismatches fail, unless snapshots are being updated.
 
-If the JSON result is different from the snapshot, the test will fail. All tests are dispatched in parallel, and the results are processed as they come in. When a result is available, it will compare the JSON result with a previously saved snapshot.
+5. **Final Summary**
+  - A pass/fail summary appears. If any test fails, the script exits with a non-zero code.
 
-### Regenerating snapshots
-To re-generate the Snapshots, run `php QitSelfTests.php update`
+---
 
-### Running only a single test type
-To run only a single test type, run `php QitSelfTests.php run TEST_TYPE`. For example, to run only the `activation` tests, run `php QitSelfTests.php run activation`. You can also `update` instead of `run`.
+## Directory Layout
 
-### How to create a new test
-- Create a new directory with the test type, eg: `performance`.
-- Create a new directory for the test, eg: `performance/test-1`.
-- Create a new `env.php` file with the parameters for the test, eg: `performance/test-1/env.php`. Copy the file format from existing tests.
-- Create the SUT plugin directory with the plugin files to test, eg: `performance/test-1/{SLUG}`, where `{SLUG}` follows the same convention as existing tests.
+```
+/home/lucas/automattic/qit-cli/
+├── qit            (the QIT binary)
+└── _tests/
+    └── managed_tests/
+        ├── QITSelfTests.php        (main script)
+        ├── src/
+        │   ├── Context.php
+        │   ├── Logger.php
+        │   ├── Config.php
+        │   ├── Validator.php
+        │   ├── TestManager.php
+        │   ├── ZipManager.php
+        │   ├── PhpUnitRunner.php
+        │   ├── QitRunner.php
+        │   ├── QITLiveOutput.php
+        │   └── test-result-parser.php
+        ├── activation/
+        │   ├── test-1/
+        │   │   ├── env.php
+        │   │   ├── ...
+        │   └── test-2/
+        │       └── env.php
+        ├── security/
+        │   └── scenario-a/
+        │       └── env.php
+        └── woo-e2e/
+            ├── scenario-1/
+            └── scenario-2/
+```
 
-That's it, now you can run `php QitSelfTests.php run performance` to run the test. Validate that the snapshot is correct, and commit the snapshot.
+- **`QITSelfTests.php`** is the primary entry point for running or updating tests.
+- Each **test type** directory (e.g., `activation/`) contains one or more subdirectories, each representing a scenario (with an `env.php` and SUT files).
 
-## Running the self-tests
+---
 
-- `./vendor/bin/phpunit ./tests/self_test`
-- Or `./vendor/bin/phpunit -d --update-snapshots ./tests/self_test`
\ No newline at end of file
+## Running Tests
+
+1. **Navigate** to `/home/lucas/automattic/qit-cli/_tests/managed_tests`.
```

Comment:
Directory change recommendation:

```suggestion
1. **Navigate** to `_tests/managed_tests`.
```

Reply:
I won't comment on all of the instances here, but I would recommend we just use the `qit-li` as the current working directory and drop `/home/lucas/automattic/` as it could be confusing if they need that directory structure or not.

Reply:
Done, thanks!